src/Components/Forms/RegisterForm.tsx

```diff
@@ -1,10 +1,44 @@
 import { useFormik } from "formik";
-import { useEffect, useState } from "react";
-import { Link } from "react-router-dom";
+import { useContext, useEffect, useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
 import * as Yup from "yup";
 import TextInput from "../Elements/TextInput";
+import axios from "axios";
+import { baseURL } from "../../const/const";
+import UserContext from "../../contexts/UserContext";
 const RegisterForm = () => {
+  axios.defaults.withCredentials = true;
   const [error, setError] = useState("");
+  const [terms, setTerms] = useState(false);
+  const { setUser } = useContext(UserContext);
+  const navigator = useNavigate();
+  const fsubmit = async (values: {
+    first_name: string;
+    last_name: string;
+    phone_number: string;
+    email: string;
+    password: string;
+    confirm_password: string;
+  }) => {
+    const result = await axios.post(baseURL + "/api/v1/auth/register", values);
+    if (result.status === 200) {
+      setUser(result.data);
+      navigator("/otp");
+    } else {
+      setError(result.data.message);
+    }
+  };
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (formik.values.password !== formik.values.confirm_password) {
```

Comment:
This can be achieved with yup validation, Give it a try but this way is also fine too

Reply:
Noted. Yup implemented.